protocols/delegate-manager/contracts/DelegateManager.sol

```diff
@@ -0,0 +1,121 @@
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+
+contract DelegateManager is Ownable {
+
+    IDelegateFactory public factory;
+    IDelegate public delegate;
+
+    /**
+      * @notice Constructor to create the Delegate Manager
+      * @dev factory is unable to be changed after the DelegateManager is constructed
+      * @param _factory address of the factory contract that will create the delegates
+      * @param _tradeWallet the wallet that the delegate will be trading on behalf of
+      */
+    constructor(IDelegateFactory _factory, address _tradeWallet) public {
+        factory = _factory;
+        delegate = IDelegate(factory.createDelegate(msg.sender, _tradeWallet));
+    }
+
+    /**
+      * @notice sets a rule on the delegate and an intent on the indexer
+      * @dev delegate needs the manager to be an admin in order to act with it.
+      * @dev manager needs to be given allowance from msg.sender for the _intent.amount
+      * @dev delegate swap needs to be given permission to move funds from the manager
+      * @param _senderToken the token the delgeate will send
+      * @param _senderToken the token the delegate will receive 
+      * @param _rule the rule to set on a delegate
+      * @param _amountToStake the amount to stake for an intent
+      * @param _indexer the indexer the intent will be set on
+      */
+    function setRuleAndIntent(
+      address _senderToken,
+      address _signerToken,
+      Types.Rule calldata _rule,
+      uint256 _amountToStake,
+      IIndexer _indexer
+    ) external onlyOwner {
+      
+      delegate.setRule(
+        _senderToken,
+        _signerToken,
+        _rule.maxSenderAmount, 
+        _rule.priceCoef,
+        _rule.priceExp
+      );
+
+      require(
+        IERC20(_indexer.stakeToken())
+        .allowance(msg.sender, address(this)) >= _amountToStake, "ALLOWANCE_FUNDS_ERROR"
+      );
+      require(
+        IERC20(_indexer.stakeToken())
+        .transferFrom(msg.sender, address(this), _amountToStake), "TRANSFER_FUNDS_ERROR"
+      );
+
+      //ensure that the indexer can pull funds from manager's account
+      require(
+        IERC20(_indexer.stakeToken())
+        .approve(address(_indexer), _amountToStake), "APPROVAL_ERROR"
+      );
+
+      _indexer.setIntent(
+        _signerToken,
+        _senderToken,
+        _amountToStake,
+       bytes32(uint256(address(delegate)) << 96) //NOTE: this will pad 0's to the right
+      );
+    }
+
+    /**
+      * @notice unsets a rule on the delegate and removes an intent on the indexer
+      * @dev delegate needs the manager to be an admin in order to act with it.
+      * @param _senderToken the maker token in the token pair for rules and intents
+      * @param _signerToken the taker token  in the token pair for rules and intents
+      * @param _indexer the indexer to remove the intent from
+      */
+    function unsetRuleAndIntent(
```

Comment:
Just been thinking... we probably also want a function to change your stake. If someone has a rule for a market, and has staked 50 tokens on that market. If they then want to increase their stake to 100, currently theyd have to call `unsetRuleAndIntent`, which would delete the rule, then they'd have to re-create the rule, and stake again.

Would be easier to have an `changeStake(token, token, newstake)` that calls unset and set for you, without deleting the rule.

Reply:
I like this idea. I will look at implementing it today.